app/components/OnboardingComponent.tsx

```diff
@@ -0,0 +1,143 @@
+'use client';
+
+import WelcomeScreen from '@/app/components/WelcomeScreen';
+import { useState } from 'react';
+import { Box } from '@mui/material';
+import OnboardingSignup from '@/app/components/OnboardingSignup';
+import OnboardingNotification from '@/app/components/OnboardingNotification';
+import OnboardingSites from '@/app/components/OnboardingSites';
+import OnboardingAddNewSite from '@/app/components/OnboardingAddNewSite';
+import { useRouter } from 'next/navigation';
+
+interface OnboardingProps {
+  dict: Dict;
+}
+
+type UserData = {
+  name: string;
+  phoneNumber: string;
+  password: string;
+};
+
+export interface UserFormErrorData extends UserData {
```

Comment:
Why does UserFormErrorData extend UserData if it already contains the same fields?

If the fields of this object are supposed to represent error-messages for each field, I would consider making them optional, so that fields without errors would have an undefined error message instead of an empty string